transport/peers/peer.go

```diff
@@ -0,0 +1,82 @@
+package peers
+
+import (
+	"go.uber.org/yarpc/transport"
+
+	"github.com/uber-go/atomic"
+)
+
+// NewPeerIdentifier creates a new HostPortPeerIdentifier
+func NewPeerIdentifier(hostport string) *HostPortPeerIdentifier {
```

Comment:
This function is specific to `HostPortPeerIdentifier`. Maybe it should be
`NewHostPortPeerIdentifier` and `NewHostPortPeer`? But that ends up being too
wordy with some amount of stuttering (`peer.NewHostPortPeer`).

What about putting the hostport implementation under `peer/hostport'? Usage
then becomes, `hostport.NewPeerIdentifier("foo:123")` and `hostport.NewPeer`.
`peer/hostportpeer` is another option (`hostportpeer.NewIdentifier` and
`hostportpeer.New`).